### ISSUE TITLE
Feature: add option to divide progress

### ISSUE THREAD

**Issue by VedmakAkaPriest** [wkh237/react-native-fetch-blob#140]
Hi ! First, I would like to express my gratitude for your library - I use it with pleasure in my project :)

As I investigate while using your library, current implementation of progress callback cause to performance issues on UI. For example, when try to download / upload a few large files in parallel.

I can suggest new feature - progress divider option for download / upload callbacks. For example, if progressDivider = 10, progress callback will be called only ten times. If progressDivider = 0 (default value), callback must be called right as it is now.

**Comment by wkh237** [comment]
@VedmakAkaPriest , thanks for the advice. I think you're right, even though we can set a debounce mechanism in JS context, there will always be an overhead to sent data from native context to JS context.  

I'll add this feature in `0.9.6`. I'm planning to add an additional argument to `uploadProgress` and `progress` so that it can be used like this 

```js
RNFetchBlob
.fetch('GET', 'http://example.com/large-one')
.uploadProgress((current, total) => {
    ...
}, 100)
.progress((current, total) => {
    ...
}, 100)
.then(() => {
  ...
})

```

How do you think ?

**Comment by VedmakAkaPriest** [comment]
It looks great! Thank you

**Comment by wkh237** [comment]
@VedmakAkaPriest , I've published a `beta` release to npm (0.9.6-beta.1), please try upgrade the package and see if it works 👍 

In this version, you can provide an object as 1st argument of `fetch.uploadProgress` and `fetch.progress` which contains property`count` and `interval`.

```js
RNFetchBlob.fetch(method, url, headers, body)
  .progress({ interval : 200, count : 10 }, (current, total ) => { ... })
  .uploadProgress({ interval : 200, count : 10 }, (current, total ) => { ... })
```

`interval` will limit the frequency of progress event (in `ms`), and `count` is a progress divider. This change is backward compatible, if there's only one argument (function) provided to `fetch.progress` or `fetch.uploadProgress`, it uses default `count=-1` and `interval=250`.

Keep in mind that sometimes `count` will not work as expected because some server does not provide response content length, in this case, the progress event frequency is only limited by `interval`.

**Comment by VedmakAkaPriest** [comment]
@wkh237 ,  I tested this feature in my project and all works fine. 
But there is one small bug (not sure, this is not essential):  when set interval = 0 in options, it considers that interval did not given at all, and resets to default.

Thank you for your work :)
